client/components/upgrades/google-apps/google-apps-dialog/index.jsx

```diff
@@ -68,10 +69,18 @@ class GoogleAppsDialog extends React.Component {
 	}
 
 	render() {
+		if ( isGsuiteRestricted() ) {
+			this.props.onAddGoogleApps();
```

Comment:
This callback takes a cart item parameter: https://github.com/Automattic/wp-calypso/blob/8c1be6eea5d411c8778aab487be1d61fb2af7b93/client/my-sites/checkout/gsuite-nudge/index.jsx#L48

Without it, it fails:
```
Uncaught TypeError: Cannot read property 'extra' of undefined
    at Object.eval [as onAddGoogleApps] (index.jsx:106)
    at GoogleAppsDialog.render (index.jsx:188)
    at finishClassComponent (react-dom.development.js:14299)
    at updateClassComponent (react-dom.development.js:14262)
    at beginWork (react-dom.development.js:15087)
```

To be honest, to get into this state, I had to hack so that the check here: https://github.com/Automattic/wp-calypso/pull/28512/files#diff-a36618f44b2f6aeae8145056394c517aR38 was not taken into account.

So, maybe the solution is simply to trust the earlier flow that this view will not be reached?